src/gpu/nvidia/cudnn_matmul_executor.hpp

```diff
@@ -82,22 +85,24 @@ struct cudnn_matmul_base_exec_t {
             void *wei_scale = arg_wei_scale.get_native_pointer(ih);
             void *dst_scale = arg_dst_scale.get_native_pointer(ih);
 
-            matmul_impl_->execute(cublas_handle, cudnn_handle, weights, src,
-                    dst, bias, reorder_scratch, src_scale, wei_scale,
+            matmul_impl_->execute(cublas_handle, cudnn_handle, params, weights,
+                    src, dst, bias, reorder_scratch, src_scale, wei_scale,
                     dst_scale);
 
-            free_runtime_scratch(matmul_impl_->has_runtime_params(),
-                    cublas_handle, cuda_stream, bias_scratch_ptr);
+            if (params.has_runtime_params_) {
+                cudaStream_t streamId;
+                cublasGetStream(cublas_handle, &streamId);
+                cudaStreamSynchronize(streamId);
```

Comment:
I think using the new host task with any synchronization is undefined behavior. Should we do this synchronization conditionally?

Reply:
Rebased on the new changes and now uses the utility func `device_sync`